packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -853,10 +881,7 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
     _syncAll(gestures);
     if (!widget.excludeFromSemantics) {
       final RenderSemanticsGestureHandler semanticsGestureHandler = context.findRenderObject();
-      context.visitChildElements((Element element) {
-        final _GestureSemantics widget = element.widget;
-        widget._updateHandlers(semanticsGestureHandler);
-      });
+      _updateSemanticsForRenderObject(semanticsGestureHandler);
```

Comment:
Would you only have to do this if no SemanticsGestureDelegate has been set? (in other words: how would changing the gestures change what semantics are available when that's determined by the SemanticsGestureDelegate alone?)

Reply:
(If I understand your question correctly)
The semantics notations are also determined by the gestures owned by the detector. It also applies to `_DefaultSemanticsGestureDelegate`, for example, it checks if the detector has `TapGestureRecognizer`, and only assigns "tap" notation if true. Since this callback replaces gesture recognizer, it needs to call `_updateSemanticsForRenderObject` no matter what delegate it uses.

Reply:
Let's talk about this in person.